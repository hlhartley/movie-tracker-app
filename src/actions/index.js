export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
});

export const loginUser = (id, name) => ({
  type: 'LOGIN_USER',
  id,
  name
});

export const updateError = (message) => ({
  type: 'UPDATE_ERROR',
  message
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});

export const showPopup = (bool) => ({
  type: 'SHOW_POPUP',
  showPopup: bool
});

export const toggleFavorite = (id) => ({
  type: 'TOGGLE_FAVORITE',
  id
});
