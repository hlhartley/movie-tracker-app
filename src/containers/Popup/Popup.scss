.popup {
    line-height: 55px;
    font-size: 30px;
    font-weight: 700;
    height: 50vh;
    background-color: $brown;
    margin-top: 35px;
    width: 80%;
    color: $silver;
    border-radius: 5px;
    position: absolute;
    z-index: 10;
    top: 42vh;
    border: 3px solid $darksilver;
    opacity: .99;
}

.popup-message{
    display: flex;
    flex-flow: column;
    justify-content: center;
    align-items: center;
    margin-top: 30px;
    text-align: center;
}

.fa-times-circle {
    margin-left: 12px;
    cursor: pointer;
    &:hover{
        color: black;
    }
}