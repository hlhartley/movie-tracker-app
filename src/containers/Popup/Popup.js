import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { showPopup } from '../../actions';
import PropTypes from 'prop-types';

export const Popup = (props) => {

  return(
    <div className='popup'>
      <i className="far fa-times-circle" onClick={() => props.showPopup(false)}></i>
      
      <div className='popup-message'>
        <p>Create a free account to save your favorite movies</p>
        <Link to='/create-account'>
          <button onClick={() => props.showPopup(false)} className='create-account-btn'>Create Account</button>
        </Link>
        <Link to='/login'>
          <button onClick={() => props.showPopup(false)} className='login-btn'>Already a user? Login</button>
        </Link>
      </div>
    </div>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  showPopup: (bool) => dispatch(showPopup(bool))
});

Popup.propTypes = {
  showPopup: PropTypes.func,
}

export default connect(null, mapDispatchToProps)(Popup)