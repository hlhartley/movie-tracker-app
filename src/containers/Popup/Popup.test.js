import React from 'react';
import { shallow } from 'enzyme';
import { Popup, mapDispatchToProps } from './Popup';
import { showPopup } from '../../actions';

describe('Popup', () => {
    describe('Popup component', () => {
        let wrapper;
        const mockShowPopup = jest.fn()
        const bool = false

        beforeEach(() => {
            wrapper = shallow(<Popup showPopup={mockShowPopup}/>)
        })

        it('should match the snapshot with all data passed in', () => {
            expect(wrapper).toMatchSnapshot()
        })

        it('should call showPopup with false when Create Account is clicked', () => {
            wrapper.find('button.create-account-btn').simulate('click')
            expect(mockShowPopup).toHaveBeenCalledWith(bool)
        })

        it('should call showPopup with false when Already a user? Login is clicked', () => {
            wrapper.find('button.login-btn').simulate('click')
            expect(mockShowPopup).toHaveBeenCalledWith(bool)
        })

        it('should call showPopup with false when the x button is clicked', () => {
            wrapper.find('i').simulate('click')
            expect(mockShowPopup).toHaveBeenCalledWith(bool)
        })

    })

    describe('mapDispatchToProps', () => {
        it('should call dispatch with a showPopup action when showPopup is called', () => {
            const mockDispatch = jest.fn()
            const bool = true;
            const actionToDispatch = showPopup(bool)
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.showPopup(bool)
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
        })
    })
})
