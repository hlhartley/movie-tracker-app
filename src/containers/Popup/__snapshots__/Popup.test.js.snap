// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Popup component should match the snapshot with all data passed in 1`] = `
<div
  className="popup"
>
  <i
    className="far fa-times-circle"
    onClick={[Function]}
  />
  <div
    className="popup-message"
  >
    <p>
      Create a free account to save your favorite movies
    </p>
    <Link
      replace={false}
      to="/create-account"
    >
      <button
        className="create-account-btn"
        onClick={[Function]}
      >
        Create Account
      </button>
    </Link>
    <Link
      replace={false}
      to="/login"
    >
      <button
        className="login-btn"
        onClick={[Function]}
      >
        Already a user? Login
      </button>
    </Link>
  </div>
</div>
`;
